testing/README.md

```diff
@@ -0,0 +1,88 @@
+# JavaScript testing styleguide
+
+This guide provides guidelines for testing JavaScript code.
+Included are recommended tools, approaches to testing, and a living styleguide.
+
+## Table of contents
+
+1. [Approach and types of testing](#approach-and-types-of-testing)
+1. [Tooling](#tooling)
+  1. [Tools for a rails stack](#tools-for-a-rails-stack)
+  1. [Tools for a non-rails stack](#tools-for-a-non-rails-stack)
+1. [Assert-style syntax rules](#assert-style-syntax-rules)
+1. [Using fixtures to represent HTML](#using-fixtures-to-represent-html)
+1. [Spying, stubbing, and mocking](#spying-stubbing-and-mocking)
+1. [Testing React](#testing-react)
+1. [Resources](#resources)
+
+## Approach and types of testing
+
+Like all software testing approaches, JavaScript testing can be divided into three types: unit testing, integration testing, and functional testing.
+
+**Unit Testing** Unit testing checks whether a particular piece (unit/module) of code is working fine. The ‘unit’ is often delimited by a JavaScript class.
+
+**Integration Testing** Integration testing is performed when various units are integrated with each other to form a sub-system or a system. This mostly focuses in the design and construction of the software architecture.
+
+**Functional Testing** The software is tested for the functional requirements. This checks whether the application is behaving according to the specification.
+
+We currently have no recommended approach to integration or functional testing. But here are some tips to keep in mind:
+
+* Avoid assertions on CSS selectors as much as possible. These types of tests are often too brittle.
```

Comment:
I'm not sure about this. If your application code refers to CSS selectors, then you should attempt to test them somehow.